### DISCUSSION TITLE
Exporting to torchscript scripting fails with type error?

### DISCUSSION THREAD

**Discussion by michaellee1** [facebookresearch/detectron2#3061] [Q&A]
I'm currently using the `export_scripting` function from [export_model.py](https://github.com/facebookresearch/detectron2/blob/master/tools/deploy/export_model.py), but running into the following error when calling 
```exported_model = export_scripting(torch_model)```

```
RuntimeError: 
Tried to access nonexistent attribute or method 'get_fields' of type '__torch__.detectron2.export.torchscript_patch16.ScriptedInstances16 (of Python compilation unit at: 0x55559f44e700)'. Did you forget to initialize an attribute in __init__()?:
  File "<ipython-input-65-8160f6d570a1>", line 25
        def forward(self, inputs: Tuple[Dict[str, torch.Tensor]]) -> List[Dict[str, Tensor]]:
            instances = self.model.inference(inputs, do_postprocess=False)
            return [i.get_fields() for i in instances]
                    ~~~~~~~~~~~~ <--- HERE

```

I ran some sample inputs through my torch model to make sure that get_fields was available after inference through my torch model.
```
sample_inputs = get_sample_inputs(cfg)
instances = torch_model.inference(sample_inputs, do_postprocess=False)
instances[0].get_fields()
```
and this ran with no problem and printed the result of get_fields.

Can someone lead me to the solution here?  How can I change the type of instances to include this method?

**Comment by ppwwyyxx** [comment]
It seems you're using the latest (github master) version of `export_model.py` but an older version of detectron2. They will have to match in versions. If this does not help please include details following the ["Unexpected behaviors" issue template](https://github.com/facebookresearch/detectron2/issues/new/choose).

**Reply by michaellee1** [reply]
@ppwwyyxx Thanks for the response Yuxin - will give this a try and update this discussion, opening an issue if needed.
